src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -372,93 +407,238 @@ void Esp32WiFiManager::factory_reset(int fd)
+        ESP_ERROR_CHECK(mdns_instance_name_set(hostname_.c_str()));
+
+        if (stationStaticIP_)
+        {
+            // Stop the DHCP service before connecting, this allows us to
+            // specify a static IP address for the WiFi connection
+            LOG(INFO, "[DHCP] Stoping DHCP Client (if running).");
+            ESP_ERROR_CHECK(
+                tcpip_adapter_dhcpc_stop(TCPIP_ADAPTER_IF_STA));
+
+            LOG(INFO,
+                "[WiFi] Configuring Static IP address:\n"
+                "IP     : " IPSTR "\n"
+                "Gateway: " IPSTR "\n"
+                "Netmask: " IPSTR,
+                IP2STR(&stationStaticIP_->ip),
+                IP2STR(&stationStaticIP_->gw),
+                IP2STR(&stationStaticIP_->netmask));
+            ESP_ERROR_CHECK(
+                tcpip_adapter_set_ip_info(TCPIP_ADAPTER_IF_STA,
+                                          stationStaticIP_));
+
+            // if we do not have a primary DNS address configure the default
+            if (ip_addr_isany(&primaryDNSAddress_))
             {
-                LOG(INFO, "[WiFi] Lost connection to SSID: %s", ssid_);
-                // clear the flag that indicates we are connected to the SSID.
-                xEventGroupClearBits(wifiStatusEventGroup_, WIFI_CONNECTED_BIT);
-                // clear the flag that indicates we have an IPv4 address.
-                xEventGroupClearBits(wifiStatusEventGroup_, DHCP_GOTIP_BIT);
-
-                // Wake up the wifi_manager_task so it can clean up
-                // connections.
-                xTaskNotifyGive(wifiTaskHandle_);
+                IP4_ADDR(&primaryDNSAddress_.u_addr.ip4, 8, 8, 8, 8);
             }
+            LOG(INFO, "[WiFi] Configuring primary DNS address to: " IPSTR,
+                IP2STR(&primaryDNSAddress_.u_addr.ip4));
+            // set the primary server (0)
+            dns_setserver(0, &primaryDNSAddress_);
+        }
+        else
+        {
+            // Start the DHCP service before connecting so it hooks into
+            // the flow early and provisions the IP automatically.
+            LOG(INFO, "[DHCP] Starting DHCP Client.");
+            ESP_ERROR_CHECK(
+                tcpip_adapter_dhcpc_start(TCPIP_ADAPTER_IF_STA));
+        }
 
-            // If we are managing the WiFi and MDNS systems we need to
-            // trigger the reconnection process at this point.
-            if (manageWiFi_)
+        LOG(INFO,
+            "[WiFi] Station started, attempting to connect to SSID: %s.",
+            ssid_);
+        // Start the SSID connection process.
+        esp_wifi_connect();
+    }
+    else if (event->event_id == SYSTEM_EVENT_STA_CONNECTED)
+    {
+        LOG(INFO, "[WiFi] Connected to SSID: %s", ssid_);
+        // Set the flag that indictes we are connected to the SSID.
+        xEventGroupSetBits(wifiStatusEventGroup_, WIFI_CONNECTED_BIT);
+    }
+    else if (event->event_id == SYSTEM_EVENT_STA_GOT_IP)
```

Comment:
is this invoked also when we are in static IP mode or in AP mode?

Reply:
Yes, it is raised almost immediately after connect in that case. I can add it to the bit_mask when static IP is used so we don't block except for connect

Reply:
no that's fine. Less code is better. Just wanted to make sure it actually works :)